source/http.c

```diff
@@ -92,811 +90,266 @@ void aws_http_load_log_subject_strings(void) {
-        AWS_HTTP_CHECK_CODE(PROXY_AUTHENTICATE);
-        AWS_HTTP_CHECK_CODE(PROXY_AUTHORIZATION);
-        AWS_HTTP_CHECK_CODE(RANGE);
-        AWS_HTTP_CHECK_CODE(REFERRER);
-        AWS_HTTP_CHECK_CODE(REFRESH);
-        AWS_HTTP_CHECK_CODE(RETRY_AFTER);
-        AWS_HTTP_CHECK_CODE(SERVER);
-        AWS_HTTP_CHECK_CODE(SET_COOKIE);
-        AWS_HTTP_CHECK_CODE(STRICT_TRANSPORT_SECURITY);
-        AWS_HTTP_CHECK_CODE(TRANSFER_ENCODING);
-        AWS_HTTP_CHECK_CODE(UPGRADE);
-        AWS_HTTP_CHECK_CODE(USER_AGENT);
-        AWS_HTTP_CHECK_CODE(VARY);
-        AWS_HTTP_CHECK_CODE(VIA);
-        AWS_HTTP_CHECK_CODE(WWW_AUTHENTICATE);
-        default:
-            return "UNKNOWN";
-    }
+static void s_headers_init(struct aws_allocator *alloc) {
+    s_header_enum_to_str[AWS_HTTP_HEADER_TRANSFER_ENCODING] = aws_byte_cursor_from_c_str("transfer-encoding");
+    s_header_enum_to_str[AWS_HTTP_HEADER_CONTENT_LENGTH] = aws_byte_cursor_from_c_str("content-length");
+    s_header_enum_to_str[AWS_HTTP_HEADER_EXPECT] = aws_byte_cursor_from_c_str("expect");
+
+    s_init_case_insensitive_hash_table(
+        &s_header_str_to_enum, alloc, s_header_enum_to_str, AWS_HTTP_HEADER_UNKNOWN + 1, AWS_HTTP_HEADER_COUNT);
 }
 
-#undef AWS_HTTP_CHECK_CODE
-
-const char *aws_http_method_to_str(enum aws_http_method method) {
-    switch (method) {
-        case AWS_HTTP_METHOD_UNKNOWN:
-            return "UNKNOWN";
-        case AWS_HTTP_METHOD_CONNECT:
-            return "CONNECT";
-        case AWS_HTTP_METHOD_DELETE:
-            return "DELETE";
-        case AWS_HTTP_METHOD_GET:
-            return "GET";
-        case AWS_HTTP_METHOD_HEAD:
-            return "HEAD";
-        case AWS_HTTP_METHOD_OPTIONS:
-            return "OPTIONS";
-        case AWS_HTTP_METHOD_PATCH:
-            return "PATCH";
-        case AWS_HTTP_METHOD_POST:
-            return "POST";
-        case AWS_HTTP_METHOD_PUT:
-            return "PUT";
-        case AWS_HTTP_METHOD_TRACE:
-            return "TRACE";
-        default:
-            return NULL;
-    }
+static void s_headers_clean_up() {
+    aws_hash_table_clean_up(&s_header_str_to_enum);
 }
 
-const char *aws_http_version_to_str(enum aws_http_version version) {
-    switch (version) {
-        case AWS_HTTP_VERSION_UNKNOWN:
-            return "UNKNOWN";
-        case AWS_HTTP_VERSION_1_0:
-            return "1.0";
-        case AWS_HTTP_VERSION_1_1:
-            return "1.1";
-        case AWS_HTTP_VERSION_2_0:
-            return "2.0";
-        default:
-            return NULL;
+enum aws_http_header_name aws_http_str_to_header_name(struct aws_byte_cursor cursor) {
+    struct aws_hash_element *elem;
+    aws_hash_table_find(&s_header_str_to_enum, &cursor, &elem);
+    if (elem) {
+        enum aws_http_header_name header = (enum aws_http_header_name)elem->value;
+        return header;
     }
+    return AWS_HTTP_HEADER_UNKNOWN;
 }
 
-const char *aws_http_code_to_str(enum aws_http_code code) {
```

Comment:
We'll probably want to eventually expose status codes as named constants to consumers

Reply:
I was debating this, I'm not sure if we actually do want to expose our constants, because the languages will need constants too, and those will need to be native, rendering our constants useless.

Reply:
It seemed like a lot of effort to curate a list of enums for common values, but also the spec says the user can invent any value they want. Then go through the effort of mirroring all those values in each CRT.

That's why I removed the enums from the public API.

Reply:
We could always add them later to the CRTs, in language-specific ways. Like, in python you have enums that natively convert to strings. Perfect for status_code->status_text.